torchrl/objectives/ddpg.py

```diff
@@ -107,35 +193,56 @@ def _forward_value_estimator_keys(self, **kwargs) -> None:
         if self._value_estimator is not None:
             self._value_estimator.set_keys(
                 value=self._tensor_keys.state_action_value,
+                reward=self._tensor_keys.reward,
+                done=self._tensor_keys.done,
             )
 
-    def forward(self, input_tensordict: TensorDictBase) -> TensorDict:
+    @property
+    def in_keys(self):
+        keys = [
+            ("next", self.tensor_keys.reward),
+            ("next", self.tensor_keys.done),
```

Comment:
this is a tricky one but the reward key can be nested and `("next", ("other", "key"))` will not work.
I usually check if the key is a str or tuple and act accordingly. We had a private discussion regarding having tensordict working with arbitrarily nested keys but did not reach a consensus yet...

Reply:
Ok, however I would need to adopt way the advantage module allows to configure the tensordict keys in the first place. Currently, the advantage module use some keys like this:
```python
tensordict.set(("next", self.tensor_keys.reward), reward)
```
which would fail if the user provides the reward key as tuple.

Reply:
Solved by unraveling keys in tensordict librarby (https://github.com/pytorch-labs/tensordict/commit/d2edd94bd7b03b508d096fa15d961064d704f638).